src/plugins/tangorin.rs

```diff
@@ -0,0 +1,195 @@
+use std::io;
+use irc::client::prelude::*;
+use regex::Regex;
+use plugin::Plugin;
+
+extern crate kuchiki;
+use kuchiki::traits::*;
+
+register_plugin!(Tangorin);
+
+lazy_static! {
+    static ref RE: Regex = Regex::new(r"!tangorin (\S+)").unwrap();
+}
+
+macro_rules! get_text {
+    ( $( $maybe_text: expr);* ) => {
+       match $($maybe_text)* {
+         Some(text) => text,
+         None => { return Ok(()); }
+       };
+    }
+}
+
+impl Tangorin {
+    fn grep_kanji(&self, msg: &str) -> Option<String> {
+        match RE.captures(msg) {
+            Some(captures) => captures.at(1).map(|e| e.to_owned()),
+            None => None
+        }
+    }
+
+    fn retrieve_from_selector(&self, doc: &kuchiki::NodeRef, selector: &str) -> Option<String> {
+        match doc.select(selector).unwrap().next() {
+            Some(match_) => {
+                let node = match_.as_node().first_child().unwrap();
+                let borrowed_text = node.as_text().unwrap().borrow();
+                Some(borrowed_text.clone().trim().to_string())
+            },
+            None => None
+        }
+    }
+
+    fn retrieve_meaning(&self, doc: &kuchiki::NodeRef) -> Option<String> {
+        match doc.select("span[class=eng]").unwrap().next() {
+            None => None,
```

Comment:
Can you think a way to avoid this `None => None` in many places? like an early return or so